bin/strata-cli/src/errors.rs

```diff
@@ -0,0 +1,89 @@
+use std::str::FromStr;
+
+pub(crate) type BoxedInner = dyn std::error::Error + Send + Sync;
+pub(crate) type BoxedErr = Box<BoxedInner>;
+
+/// This indicates runtime failure in the underlying platform storage system. The details of the
+/// failure can be retrieved from the attached platform error.
+#[derive(Debug)]
+#[allow(unused)]
+pub struct PlatformFailure(BoxedErr);
+
+impl PlatformFailure {
+    pub fn new<E>(e: E) -> Self
+    where
+        E: Into<BoxedErr>,
+    {
+        Self(e.into())
+    }
+}
+
+/// This indicates that the underlying secure storage holding saved items could not be accessed.
+/// Typically this is because of access rules in the platform; for example, it might be that the
+/// credential store is locked. The underlying platform error will typically give the reason.
+#[derive(Debug)]
+#[allow(unused)]
+pub struct NoStorageAccess(BoxedErr);
+
+impl NoStorageAccess {
+    pub fn new<E>(e: E) -> Self
+    where
+        E: Into<BoxedErr>,
+    {
+        Self(e.into())
+    }
+}
+
+/// Errors displayed to the user when using the Strata CLI
+pub enum DisplayedError {
+    /// Errors the use can address by updating configuration or providing expected input
+    UserError(String),
+    /// Internal errors encountered when servicing user's request.
+    InternalError(String, Box<dyn std::fmt::Debug>),
+}
+
+pub(crate) fn user_error(msg: impl Into<String>) -> DisplayedError {
+    DisplayedError::UserError(msg.into())
+}
+
+#[allow(unused)]
+pub(crate) fn internal_error<E>(msg: impl AsRef<str>) -> impl FnOnce(E) -> DisplayedError
+where
+    E: std::fmt::Debug + 'static,
+{
+    move |e| {
+        DisplayedError::InternalError(
+            String::from_str(msg.as_ref()).expect("infallible"),
+            Box::new(e),
+        )
+    }
+}
+
+pub(crate) trait DisplayableError {
+    type T;
```

Comment:
This associated type should be `Output`.  Just `T` as an associated type is ambiguous.

I'm a bit confused on how this is intended to be used, when considering the names here.  For a comparison, see `anyhow`'s `context` and `with_context` functions.  It seems like this is doing something similar.